### ISSUE TITLE
closestPointTo dies silently

### ISSUE THREAD

**Issue by cal6** [tmpvar/polygon.js#6]
Current code dies silently while in this iteration (polygon.js:202):

    for (i=0; i<l; i++) {
      var a = this.point(i-1);
      var b = this.point(i);

Workaround/fix to make it work:

    for (i=0; i<l-1; i++) {
      var a = this.point(i);
      var b = this.point(i+1);

At the moment I'm not sure what this change can break but it resolved my problem when calculating closest point to a polygon.

**Comment by tmpvar** [comment]
hey, thanks for reporting.  Can you send over the polygon that you were using to arrive at this behavior?

**Comment by cal6** [comment]
Sure :)

```
var polygon = require('polygon'),
        vec2 = require('vec2');

var point = vec2(10, 10);
var pol = new polygon([
        vec2(1,1),
        vec2(5,1),
        vec2(5,5),
        vec2(1,5),
        vec2(1,1)
]);
console.log(pol.closestPointTo(point));
```

Response from npm version 1.0.1:
```
      throw new Error('NaN detected');
            ^
```

Response with my patch:
```
{ x: 5,
  y: 5,
  prev: { x: 1, y: 1 },
  next: { x: 5, y: 5 },
  current: { x: 5, y: 1 } }
```

**Comment by tmpvar** [comment]
oh, that's not really silent is it ;) - anyhow thanks for reporting I'll try to figure out why we're getting a `NaN` in this case later on today

**Comment by tmpvar** [comment]
published polygon@1.0.2 with this fix, thanks!

**Comment by cal6** [comment]
Thanks for the update! :)

The silence thing, after debugging, really did came from my code.
